pkg/controllers/operation_controller.go

```diff
@@ -116,17 +117,10 @@ func (o *OperationReconciler) ReconcileInternal(ctx dataoperation.ReconcileReque
 	}
 
 	// 2. set target dataset
-	targetDatasetNamespacedName, err := o.implement.GetTargetDatasetNamespacedName(object)
-	if err != nil {
-		ctx.Log.Error(err, "Failed to get the ddc dataset namespace and name")
-		return utils.RequeueIfError(errors.Wrap(err, "Unable to get dataset"))
-	}
-
-	targetDataset, err := utils.GetDataset(o.Client, targetDatasetNamespacedName.Name, targetDatasetNamespacedName.Namespace)
+	targetDataset, err := o.implement.GetTargetDataset(object)
 	if err != nil {
 		if utils.IgnoreNotFound(err) == nil {
-			ctx.Log.Info("The dataset is not found", "dataset", targetDatasetNamespacedName)
-			// no dataset means no metadata, not necessary to Reconcile
+			ctx.Log.Info("The dataset is not found")
```

Comment:
Suggest to add more log messages here. For example, log with the object's namespace/name and operationType

Reply:
Plus, do we need to issue an event to inform users of the missing target dataset for cases like a typo mistake when indicating the dataset name

Reply:
1. `ctx.Log` has the object type and name/namespace info.
2. fixed logging dataset name when occuring not found error.